salt/tests/unit/modules/files/test_metalk8s_kubernetes_utils.yaml

```diff
@@ -0,0 +1,89 @@
+# test case for `get_kubeconfig` function
+# This test case gets the kubeconfig and context based on the order below:
+  # 1. args
+  # 2. directly from pillars
+  # 3. from the salt-master configuration `config.option`
+get_kubeconfig:
+  # 1. get kubeconfig from kwargs options only
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf", "context": "my-kwargs-mycontext"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 2. get kubeconfig from api-server pillars only
+  - api_server_pillar: {
+      "kubeconfig": "my-pillar-kubeconfig.conf",
+      "context": "my-pillar-context"
+    }
+    result: ['my-pillar-kubeconfig.conf', 'my-pillar-context']
+  # 3. get kubeconfig from salt config options only
+  - config_options: {
+      "kubernetes.kubeconfig": "my-config-option-kubeconfig.conf",
+      "kubernetes.context": "my-config-option-mycontext"
+    }
+    result: ['my-config-option-kubeconfig.conf', 'my-config-option-mycontext']
+
+  # 4. get kubeconfig partly from kwargs and salt config options
+  - kwargs: {"context": "my-kwargs-mycontext"}
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    result: ['my-config-option-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 5. get kubeconfig partly from kwargs and api-server pillars
+  - kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    api_server_pillar: {"context": "my-pillar-context"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-pillar-context']
```

Comment:
As I already told you, this one does not make sens the context in the pillar and set the kubeconfig in kwargs, do the opposite (kubeconfig in pillar and context in kwargs)